pkg/util/azureclient/azuresdk/armnetwork/flowlogs.go

```diff
@@ -0,0 +1,29 @@
+package armnetwork
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
+)
+
+// FlowLogsClient is a minimal interface for azure FlowLogsClient
+type FlowLogsClient interface {
+	Get(ctx context.Context, resourceGroupName string, networkWatcherName string, flowLogName string, options *armnetwork.FlowLogsClientGetOptions) (result armnetwork.FlowLogsClientGetResponse, err error)
+
+	FlowLogsClientAddons
+}
+
+type flowLogsClient struct {
+	*armnetwork.FlowLogsClient
+}
+
+func NewFlowLogsClient(subscriptionID string, credential azcore.TokenCredential, options *arm.ClientOptions) (FlowLogsClient, error) {
```

Comment:
Reference Comment: https://github.com/Azure/ARO-RP/pull/3619#discussion_r1670895639

We should be returning structs, not interfaces as per the go standard. You can refer the roledefinitions client for the usage https://github.com/Azure/ARO-RP/blob/master/pkg/util/azureclient/azuresdk/armauthorization/roledefinitions.go

For now we can ignore the exisitng clients returning interfaces, but can stop returning the interfaces for new code.

Reply:
Thanks! nice catch!